### DISCUSSION TITLE
Drivetrain Model

### DISCUSSION THREAD

**Discussion by Camille0327** [OpenFAST/openfast#2042] [Q&A]
Dear all,
    I am working on the two-mass drive train model now. Here are two problems to ask for help:
    First. For the open loop wind turbine system, I get the state space equations as shown in the following graph. (It needs to be mentioned that all the values in the low-speed shaft side has been transformed into the high-speed side.) Could you please help me check whether the state equations I derived is right or not?
    Second. For the MPPT control in the second wind turbine working region, when I choose the power closed loop control method, what new state space equations can I get, based on the open-loop state space equation I already derived? According to the control logic, I can get the expression of the electromagnetic torque Te as shown in the following graph, but I don’t know how to derive δTe, which should be expressed by all state variables. Could you please help me to choose one more new state variable, which I think can be the generator rotating angle θe, and then derive the new state equations?
![1](https://github.com/OpenFAST/openfast/assets/148426697/e20b2783-5ebb-42a5-9efd-6cb546afe45b)
    I know it must take time to solve, so I am really grateful for your patience and help! Thank you in advance for your responses!
@jjonkman @bjonkman @NRELAdmin

**Comment by jjonkman** [comment]
Duplicated in #2043.
